nmpolicy/internal/resolver/path.go

```diff
@@ -57,32 +57,33 @@ func applyFuncOnSlice(originalSlice []interface{},
 	expectedNode ast.Node,
 	funcToApply func(map[string]interface{}, string, ast.Node) (interface{}, error),
 	shouldFilterSlice bool) (interface{}, error) {
-	filteredSlice := []interface{}{}
+	adjustedSlice := []interface{}{}
+	sliceEmptyAfterApply := true
 	for _, valueToCheck := range originalSlice {
-		value, err := applyFuncOnPath(valueToCheck, path, expectedNode, funcToApply, false)
+		valueAfterApply, err := applyFuncOnPath(valueToCheck, path, expectedNode, funcToApply, false, false)
 		if err != nil {
 			return nil, err
 		}
-		if value != nil {
-			filteredSlice = append(filteredSlice, valueToCheck)
+		if valueAfterApply != nil {
+			sliceEmptyAfterApply = false
+			adjustedSlice = append(adjustedSlice, valueAfterApply)
+		} else if !shouldFilterSlice {
+			adjustedSlice = append(adjustedSlice, valueToCheck)
 		}
 	}
 
-	if len(filteredSlice) == 0 {
+	if sliceEmptyAfterApply {
```

Comment:
I would keep it `len(adjustedSlice) == 0`.

Reply:
Is no tthe same, adjustedSlice can be > 0 and sliceEmptyAfterApply false.